sys/linux/usbip.txt

```diff
@@ -0,0 +1,47 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <drivers/usb/usbip/usbip_common.h>
+
+resource usbip_server_fd[fd]
+
+syz_usbip_server_init(usb3 bool32, speed flags[usbip_device_speed]) usbip_server_fd
+write$usbip(fd usbip_server_fd, buffer ptr[in, usbip_header], len bytesize[buffer])
+
+type usbip_header_basic[PACKET_TYPE] {
+	command		const[PACKET_TYPE, int32be]
+	seqnum		int32be
+	devid		int32be
+	direction	flags[direction_flags, int32be]
+	ep		int32be
+} [packed]
+
+usbip_header_ret_submit {
```

Comment:
Can we turn this into template and pass `actual_length` and `number_of_packets` as arguments?

```
type usbip_header_ret_submit[LENGTH, PACKETS] {
...
  actual_length  LENGTH
...
  number_of_packets  PACKETS
...
}
```

And then in `usbip_packet_ret_submit` use:

```
  ret_submit  usbip_header_ret_submit[length[xbuff], length[iso]]
```